src/main/java/org/opensearch/geospatial/GeospatialParser.java

```diff
@@ -30,4 +35,34 @@
         }
         return stringObjectMap;
     }
+
+    /**
+     * User inputs are usually deserialized into Map. extractValueAsString will help caller to
+     * extract value from the Map and cast it to string with validation.
+     * @param input User input of type Map
+     * @param key property we would like to extract value of
+     * @return null if key doesn't exist, value as String if it exists, throw exception otherwise
+     */
+    public static String extractValueAsString(final Map<String, Object> input, final String key) {
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(input);
```

Comment:
can we do the same addition with descriptive user message here as well?

Reply:
Ack